app/jobs/unsubscribe_from_list_job.rb

```diff
@@ -0,0 +1,11 @@
+class UnsubscribeFromListJob < ApplicationJob
+  queue_as :default
+  sidekiq_options retry: 5
+
+  def perform(subscription)
+    return unless prospect_id = Pardot.salesforce_to_prospect(subscription.contact.salesforce_id)
+
+    Pardot.client.list_memberships.delete(subscription.list.pardot_id, prospect_id)
+    subscription.destroy!
```

Comment:
changed this to blow up if the destroy fails - so we should get a sentry error and the job will fail